### PR TITLE
TIG-3599 Update genny to use new vcpkg/toolchain

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: src/lamplib/src/genny/tasks/compile.py [hunk 1]
```diff
@@ -162,14 +166,24 @@ def compile_and_install(
...
-    compile_all(
-        genny_repo_root=genny_repo_root,
-        workspace_root=workspace_root,
-        build_system=build_system,
-        os_family=os_family,
-        linux_distro=linux_distro,
-        ignore_toolchain_version=ignore_toolchain_version,
-    )
+
+    try:
```
Comment (richardsamuels) [inline src/lamplib/src/genny/tasks/compile.py:170]: This is the new warning message that I've added

**Review by guoyr — APPROVED** [review]
LGTM! FYI there's a project to improve the toolchain rollout process, which should allow us to explicitly check for the min mongo toolchain version. (https://jira.mongodb.org/browse/PM-2366)

**Comment by richardsamuels** [comment]
Sys-perf: https://evergreen.mongodb.com/version/61ddd2cb1e2d176f03f4da05 

^ merging since that was clean.

**Comment by richardsamuels** [comment]
evergreen merge

**Pull request merged**